Plugins/Public/event/Main.cpp

```diff
@@ -703,6 +774,23 @@ void __stdcall GFGoodBuy_AFTER(struct SGFGoodBuyInfo const &gbi, unsigned int iC
 					}
 				}
 
+				// Is the event limited by Ship class?
+				if (i->second.isClassRestricted)
+				{
+					//PrintUserCmdText(iClientID, L"Checking ship class requirements....");
+					Archetype::Ship* TheShipArch = Archetype::GetShip(Players[iClientID].iShipArchetype);
+					wstring& classname = shipclassnames.find(TheShipArch->iShipClass)->second;
+					
+					if (classname != i->second.allowedShipClass)
+					{
+						PrintUserCmdText(iClientID, L"DEBUG: Invalid ship class[%s], this Event is for [%s]'s", classname.c_str(), i->second.allowedShipClass.c_str());
```

Comment:
```suggestion
						PrintUserCmdText(iClientID, L"DEBUG: Invalid ship class %s, this event is for %ss", classname.c_str(), i->second.allowedShipClass.c_str());
```

Reply:
This print should be hidden behind a debug flag or removed, we don't want players to see this message.

Reply:
We should notify players they will not be a part of the bonus event, but in a non-DEBUG method.